packages/authorization-server/src/routers/AuthorizationServerRouter.ts

```diff
@@ -61,12 +63,15 @@ const authorizationServerRouter = (
   authorizationServerRouter.post(
     "/authorization-server/token.oauth2",
     async (req, res) => {
-      const ctx = fromAppContext(req.ctx);
+      const ctx: WithLogger<AuthServerAppContext> = {
+        ...req.ctx,
+        logger: logger({ ...req.ctx }),
+      };
 
       try {
         const tokenResult = await tokenService.generateToken(
           req.body,
-          ctx.correlationId,
+          req.ctx,
           ctx.logger
```

Comment:
Here we usually pass the entire context, the fact that you pass just `ctx.logger`, don't pass the `ctx` you created above that includes the logger, but then pass `req.ctx` instead, makes this look misleading. 

If it's because you mutate it to add clientId/organizationId into the req.ctx, I propose adding these two values to the generateToken return value and setting req.ctx.clientId / req.ctx.organizationId here in the router

```suggestion
          ctx
```

Reply:
Thanks